ui/src/main.rs

```diff
@@ -72,10 +79,45 @@ fn main() {
     let logger = StatisticLogger::new(file_logger);
     chain.link_around(logger);
 
+    if env::var("CORS").is_ok() {
+        debug!("using CORS");
+        chain.link_around(CorsMiddleware);
+        chain.link_after(CorsMiddleware);
+    }
+
     info!("Starting the server on {}:{}", address, port);
     Iron::new(chain).http((&*address, port)).expect("Unable to start server");
 }
 
+struct CorsMiddleware;
+
+impl AroundMiddleware for CorsMiddleware {
+    fn around(self, handler: Box<Handler>) -> Box<Handler> {
+        Box::new(move | req: &mut Request | {
+            match req.method {
+                Method::Options => Ok(Response::with((status::Ok))),
+                _ => handler.handle(req)
+            }
+        })
+    }
+}
+
+impl AfterMiddleware for CorsMiddleware {
+    fn after(&self, _: &mut Request, mut resp: Response) -> IronResult<Response> {
+        resp.headers.set(AccessControlAllowOrigin::Value("*".to_string()));
+        let headers = vec![
+                    UniCase("Origin".to_owned()),
+                    UniCase("X-Requested-With".to_owned()),
+                    UniCase("Content-Type".to_owned()),
+                    UniCase("Accept".to_owned()),
+                    ];
+        resp.headers.set(AccessControlAllowHeaders(headers));
+        let methods = vec![ Method::Get, Method::Put, Method::Post ];
+        resp.headers.set(AccessControlAllowMethods(methods));
+        Ok(resp)
```

Comment:
We should consider adding `Access-Control-Max-Age` with 1 hour or so, to avoid the browser having to do the preflight checks over and over.